### PR TITLE
Aes crash fix

### PR DESCRIPTION
**Pull request by elv-reza**
- This avoids a crash that happens time to time on some of the servers. 
- Usually the crash happens with split transcoding that generates two slices of segments.

### CODE REVIEW

File: libavformat/dashenc.c [hunk 1]
```diff
@@ -698,6 +698,12 @@ static void dash_free(AVFormatContext *s)
             else
                 avio_close(os->ctx->pb);
         }
+        /*
+         * ff_format_io_close() releases and sets os->out to NULL,
+         * so aes_free() should be done before ff_format_io_close() to
+         * avoid a crash in avio_write(os->out, ...).
+         */
+        aes_free(os);
```
Comment (elv-serban) [inline libavformat/dashenc.c:706]: What does aes_free do? Frees an object inside os?

**Review by elv-serban — APPROVED** [review]
Looks good

File: libavformat/dashenc.c (hunk 1 above)
Comment (elv-reza) [reply libavformat/dashenc.c:706]: FYI,
```
static void aes_free(OutputStream *os) {
    if (!os->aes_encrypt) return;
    if (os->aes_context) {
        uint8_t out_buf[BLOCKSIZE];
        int pad = BLOCKSIZE - os->aes_pad_len;

        memset(&os->aes_pad[os->aes_pad_len], pad, pad);
        av_aes_crypt(os->aes_context, out_buf, os->aes_pad, 1, os->aes_iv, 0);
        avio_write(os->out, out_buf, BLOCKSIZE);
    }
    av_freep(&os->aes_context);
    av_freep(&os->aes_write_buf);
    os->aes_encrypt = 0;
    os->aes_pad_len = 0;
    os->aes_write_buf_size = 0;
}

```

Comment (elv-peter) [reply libavformat/dashenc.c:706]: I'd add an assert or check for a valid `os->out` in `aes_free()`. Also clarify in the comment that `aes_free()` calls `avio_write()`.

**Pull request merged**